bigquery/src/test/java/org/apache/iceberg/gcp/bigquery/TestBigQueryCatalog.java

```diff
@@ -0,0 +1,673 @@
+        .isEqualTo(TABLE_SPEC.fields());
+    assertThat(table.sortOrder())
+        .as("Table should have create sort order")
+        .isEqualTo(TABLE_WRITE_ORDER);
+    assertThat(table.properties().entrySet())
+        .as("Table properties should be a superset of the requested properties")
+        .containsAll(properties.entrySet());
+    if (!overridesRequestedLocation()) {
+      assertThat(table.location())
+          .as("Table location should match requested")
+          .isEqualTo(tableLocation);
+    }
+    assertFiles(table, FILE_A);
+    assertFilesPartitionSpec(table);
+    assertPreviousMetadataFileCount(table, 0);
+  }
+
+  @Disabled(
+      "BigQuery does not allow adding a REQUIRED column to an existing table schema. "
+          + "When this operation is attempted, an exception gets thrown.")
+  @Test
+  public void testCreateOrReplaceReplaceTransactionReplace() {}
+
+  // TODO: Remove this test when BigQuery supports required field change
+  @Test
+  public void testCreateOrReplaceReplaceTransactionReplaceWithOptionalFields() {
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(NS);
+    }
+
+    Table original = catalog.buildTable(TABLE, OTHER_SCHEMA_OPTIONAL).create();
+
+    assertThat(catalog.tableExists(TABLE))
+        .as("Table should exist before replaceTransaction")
+        .isTrue();
+
+    Transaction createOrReplace =
+        catalog.buildTable(TABLE, SCHEMA_OPTIONAL).createOrReplaceTransaction();
+
+    assertThat(catalog.tableExists(TABLE))
+        .as("Table should still exist after replaceTransaction")
+        .isTrue();
+
+    createOrReplace.newFastAppend().appendFile(FILE_A).commit();
+
+    // validate table has not changed
+    Table table = catalog.loadTable(TABLE);
+
+    assertThat(table.schema().asStruct())
+        .as("Table schema should match concurrent create")
+        .isEqualTo(OTHER_SCHEMA_OPTIONAL.asStruct());
+
+    assertUUIDsMatch(original, table);
+    assertNoFiles(table);
+
+    createOrReplace.commitTransaction();
+
+    // validate the table after replace
+    assertThat(catalog.tableExists(TABLE)).as("Table should exist after append commit").isTrue();
+    table.refresh(); // refresh should work with UUID validation
+
+    Table loaded = catalog.loadTable(TABLE);
+
+    assertThat(loaded.schema().asStruct())
+        .as("Table schema should match the new schema")
+        .isEqualTo(REPLACE_SCHEMA_OPTIONAL.asStruct());
+    assertUUIDsMatch(original, loaded);
+    assertFiles(loaded, FILE_A);
+    assertPreviousMetadataFileCount(loaded, 1);
+  }
+
+  @Disabled(
+      "BigQuery does not allow adding a REQUIRED column to an existing table schema. "
+          + "When this operation is attempted, an exception gets thrown.")
+  @Test
+  public void testConcurrentReplaceTransactionSchema2() {}
+
+  // TODO: Remove this test when BigQuery supports required field change
+  @Test
+  public void testConcurrentReplaceTransactionSchema2WithOptionalFields() {
```

Comment:
I don't think any of these tests that use optional fields are needed here. The goal here is to verify all of the functionality at the catalog level

Reply:
These tests were also run against the Production BigQuery environment using the real BigQuery client. This helped identify a limitation within BigQuery itself. To work around this limitation, I've implemented optional field tests.

Reply:
Removed all optionalField tests